src/agent/metrics.rs

```diff
@@ -0,0 +1,376 @@
+            })
+            .await?;
+
+        self.global_store_lookup_tx
+            .send(Lookup::LookupAllAccountsData {
+                result_tx: global_data_tx,
+            })
+            .await?;
+
+        self.global_store_lookup_tx
+            .send(Lookup::LookupAllAccountsMetadata {
+                result_tx: global_metadata_tx,
+            })
+            .await?;
+
+        // Await the results
+        let local_data = local_rx.await?;
+        let global_data = global_data_rx.await??;
+        let global_metadata = global_metadata_rx.await??;
+
+        let symbol_view =
+            build_dashboard_data(local_data, global_data, global_metadata, &self.logger);
+
+        let mut uptime_seconds = self.start_time.elapsed().as_secs();
+
+        // Shave bigger units off the seconds value. Decide the count
+        // and replace with remainder.
+        let uptime_days = uptime_seconds / (24 * 3600); // 24h
+        uptime_seconds %= 24 * 3600;
+
+        let uptime_hours = uptime_seconds / 3600; // 1h
+        uptime_seconds %= 3600;
+
+        let uptime_minutes = uptime_seconds / 60; // 1min
+        uptime_seconds %= 60;
+
+        let uptime_string = format!(
+            "{}d{}h{}m{}s",
+            uptime_days, uptime_hours, uptime_minutes, uptime_seconds
+        );
+
+        // Build and collect table rows
+        let mut rows = vec![];
+
+        for (symbol, data) in symbol_view {
+            for (price_pubkey, price_data) in data.prices {
+                let price_string = if let Some(global_data) = price_data.global_data {
+                    let expo = global_data.expo;
+                    let price_with_expo: f64 = global_data.agg.price as f64 * 10f64.powi(expo);
+                    format!("{:.2}", price_with_expo)
+                } else {
+                    "no data".to_string()
+                };
+
+                let last_publish_string = if let Some(global_data) = price_data.global_data {
+                    if let Some(datetime) =
+                        NaiveDateTime::from_timestamp_opt(global_data.timestamp, 0)
+                    {
+                        datetime.format("%Y-%m-%d %H:%M:%S").to_string()
+                    } else {
+                        format!("Invalid timestamp {}", global_data.timestamp)
+                    }
+                } else {
+                    "no data".to_string()
+                };
+
+                let last_local_update_string = if let Some(local_data) = price_data.local_data {
+                    if let Some(datetime) =
+                        NaiveDateTime::from_timestamp_opt(local_data.timestamp, 0)
+                    {
+                        datetime.format("%Y-%m-%d %H:%M:%S").to_string()
+                    } else {
+                        format!("Invalid timestamp {}", local_data.timestamp)
+                    }
+                } else {
+                    "no data".to_string()
+                };
```

Comment:
nitpik: maybe in the future I move these to another method to separate the html rendering vs what data to display.

Reply:
I already broke ground on the metrics front, starting with rearranging the modules. It definitely looks better that way and is coming soon.